libraries/helpers.rb

```diff
@@ -27,11 +27,11 @@ def register_command
         command << activation_keys.map { |key| "--activationkey=#{key}" }
         command << "--org=#{organization}"
       elsif username && password
-        raise 'Unable to register - must specify environment when using username/password' if environment.nil?
+        raise 'Unable to register - must specify environment when using username/password' if environment.nil? && using_satellite_host?
 
-        command << "--username=#{username}"
-        command << "--password=#{password}"
-        command << "--environment=#{environment}"
+        command << "--username='#{username}'"
```

Comment:
`Shellwords` has a `shellescape` method.  If that's the problem you're trying to solve then I'd use it.